autopilot/agent.go

```diff
@@ -473,17 +475,33 @@ func (a *Agent) controller() {
 		pendingMtx.Unlock()
 
 		// Now that we've updated our internal state, we'll consult our
-		// channel attachment heuristic to determine if we should open
+		// channel attachment heuristics to determine if we should open
 		// up any additional channels or modify existing channels.
-		availableFunds, numChans, needMore := a.cfg.Heuristic.NeedMoreChans(
-			totalChans, a.totalBalance,
-		)
-		if !needMore {
+		var heuristic AttachmentHeuristic
```

Comment:
IMO the constraints should be lifted to the agent level. So instead, this loop would query all heuristics for their top candidates, and then (to start something simple) sum the scores from each heuristic, and use that as a basis (along with jitter) to select a possible candidate.